### ISSUE TITLE
Choose Validation code from MDETR repo and adjust input data accordingly

### ISSUE THREAD

**Issue by shivahanifi** [shivahanifi/GazeMDETR#17]
Check the instructions on the MDETR repo to find the proper validation code.
1. Adapt the test data to the evaluation code
2. Make sure to use the right model (how to call the model, etc.)

**Comment by shivahanifi** [comment]
There are 3 different validation codes available in the [MDETR repo](https://github.com/ashkamath/mdetr/tree/main/scripts).

|Parameters/Evaluation code|[eval_clevr.py](https://github.com/ashkamath/mdetr/blob/main/scripts/eval_clevr.py)|[eval_gqa.py](https://github.com/ashkamath/mdetr/blob/main/scripts/eval_gqa.py)|[eval_lvis.py](https://github.com/ashkamath/mdetr/blob/main/scripts/eval_lvis.py)|
|---|---|---|---|
|Notes|This script allows to dump the model's prediction on an arbitrary split of [CLEVR](https://cs.stanford.edu/people/jcjohns/clevr/)/[CoGenT](https://autonlp.ai/datasets/a-diagnostic-dataset-for-compositional-language-and-elementary-visual-reasoning-(clevr-&-cogent))/[CLEVR-Humans](https://cs.stanford.edu/people/jcjohns/iep/)|||
|get_coco_api_from_dataset|NO| YES|YES not used|
|dataset specific import|datasets.clevr|NO|datasets.lvis_eval|

**Comment by shivahanifi** [comment]
using the GazeMDETR_eval.py file try to calculate the iou and recall. (hints from flickr_eval.py)

- added the iou calculation to the demo code
   - Find out how to save them separately and also compute the whole iou accuracy by specifying the iou thresh
   -

**Comment by shivahanifi** [comment]
## important parts from the paper:

- We evaluate our method on 4 downstream tasks: referring expression comprehension and segmentation, visual question answering and phrase grounding
    -  **Phrase grounding:** Given one or more phrases, which may be inter-related, the task is to provide a set of bounding boxes for each phrase. We use the Flickr30k entities dataset for this task, with the train/val/test splits and evaluate our performance in terms of Recall@k. For each sentence in the test set, we predict 100 bounding boxes and use the soft token alignment prediction to rank the boxes according to the score given to the token positions that correspond to the phrase. We evaluate under two protocols which we name ANY-BOX  and MERGED-BOXES.
    - **Referring expression comprehension:** Given an image and a referring expression in plain text, the task is to localize the object being referred to by returning a bounding box around it.  we train our model to directly predict the bounding box, given a referring expression and the associated image. There are three established datasets for this task called RefCOCO, RefCOCO+  and RefCOCOg. Since during pre-training we annotate every object referred to within the text, there is a slight shift in the way the model is used in this task. For example, during pre-training, given the caption “The woman wearing a blue dress standing next to the rose bush.”, MDETR would be trained to predict boxes for all referred objects such as the woman, the blue dress and the rose bush. However, for referring expressions, the task would be to only return one bounding box, which signifies the woman being referred to by the entire expression. For this reason, we finetune the model on the task-specific. dataset for 5 epochs

**Comment by shivahanifi** [comment]
## Evaluation metric to be used for GazeMDETR

Considering the information mentioned above, the recall metric will be used. However, since with the GazeMDETR we aim to have only one bounding box as an output (choosing the one with the highest confidence level)  only recall@1 will be considered with the adjustable IoU threshold.

- from the `flickr_eval.py` the computations for recall@k can be extracted:
   
     ```
                ious = box_iou(np.asarray(cur_boxes), np.asarray(target_boxes))
                for k in self.topk:
                    maxi = 0
                    if k == -1:
                        maxi = ious.max()
                    else:
                        assert k > 0
                        maxi = ious[:k].max()
                    if maxi >= self.iou_thresh:
                        recall_tracker.add_positive(k, "all")
                        for phrase_type in phrase["phrase_type"]:
                            recall_tracker.add_positive(k, phrase_type)
                    else:
                        recall_tracker.add_negative(k, "all")
                        for phrase_type in phrase["phrase_type"]:
                            recall_tracker.add_negative(k, phrase_type)

    ```
